src/Endatix.Core/UseCases/FormDefinitions/PartialUpdateActive/PartialUpdateActiveFormDefinitionHandler.cs

```diff
@@ -18,10 +19,24 @@ public async Task<Result<FormDefinition>> Handle(PartialUpdateActiveFormDefiniti
             return Result.NotFound("Active form definition not found.");
         }
 
-        activeDefinition.UpdateSchema(request.JsonData);
-        activeDefinition.UpdateDraftStatus(request.IsDraft);
+        if(await FormDefinitionHasSubmissions(activeDefinition.Id) && activeDefinition.JsonData != request.JsonData) {
+            var newFormDefinition = new FormDefinition(jsonData: request.JsonData);
+            await formRepository.AddNewFormDefinitionAsync(formWithActiveDefinition, newFormDefinition, cancellationToken);
+            newFormDefinition.UpdateDraftStatus(request.IsDraft);
+            formWithActiveDefinition.SetActiveFormDefinition(newFormDefinition);
+        }
+        else {
+            activeDefinition.UpdateSchema(request.JsonData);
+            activeDefinition.UpdateDraftStatus(request.IsDraft);
+        }
 
         await formRepository.UpdateAsync(formWithActiveDefinition, cancellationToken);
         return Result.Success(activeDefinition);
     }
+
+    private async Task<bool> FormDefinitionHasSubmissions(long definitionId) {
+        var spec = new SubmissionsTotalCountByFormDefinitionIdSpec(definitionId);
+        var count = await submissionsRepository.CountAsync(spec);
```

Comment:
Can you rewrite this with `AnyAsync()`. It is recommended for performance when we don't need the count, but if there are items at all - https://admirmujkic.medium.com/why-the-any-method-can-be-faster-than-count-in-c-9921a092a492